frame/support/src/traits.rs

```diff
@@ -192,6 +192,15 @@ macro_rules! impl_filter_stack {
 	}
 }
 
+/// Type that provide some integrity tests.
+///
+/// This implemented for modules by `decl_module`.
+#[impl_for_tuples(30)]
+pub trait IntegrityTest {
+	/// Run integrity test.
```

Comment:
```suggestion
	/// Run integrity test.
	///
	/// The test is not executed in a externalities provided environment.
```

Maybe we should add this?